### PR TITLE
fix: Fix exceptions that occur when adding nodes back to the host clu…

### LINKED ISSUE #631: kubenest unjoin problem
**What happened**:

When a node is sold and then returned, that is, when the node's status changes from occupied to free, there is an issue with the pod's status scheduled to the free node.
6001718936376_ pic_副本
6021718936376_ pic_副本

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kosmos version:
- Others:

### PR DESCRIPTION
**Pull request by OrangeBao**
…ster

#### What type of PR is this?

/kind bug

#### Which issue(s) does this PR fix?

Fixes #631 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/k8s-in-k8s/generate_env.sh [hunk 1]
```diff
@@ -68,6 +68,10 @@ function GetKubernetesCaPath() {
     kubectl get cm kubelet-config -nkube-system -oyaml  | awk '/clientCAFile:/{print $2}'
 }
 
+function GetKubeDnsClusterIP() {
+    kubectl get svc -nkube-system kube-dns  -o jsonpath='{.spec.clusterIP}'
```
Comment (duanmengkk) [inline hack/k8s-in-k8s/generate_env.sh:72]: kubectl may be in other path such as ”/home/deploy/kubectl“ in some env

File: hack/k8s-in-k8s/generate_env.sh [hunk 2]
```diff
@@ -135,6 +142,41 @@ nodeRegistration:
...
+      readOnly: true
+  hostNetwork: true
+  priorityClassName: system-node-critical
+  volumes:
+  - hostPath:
+      path: /apps/conf/nginx
+      type: 
+    name: etc-nginx
+status: {}\" > $PATH_KUBERNETES/manifests/nginx-proxy.yaml
+}
```
Comment (duanmengkk) [inline hack/k8s-in-k8s/generate_env.sh:178]: This feature can be looked into later on how it's implemented with kubekey, introducing an nginx seems unnecessary.

Comment (OrangeBao) [reply hack/k8s-in-k8s/generate_env.sh:178]: ok

File: hack/k8s-in-k8s/generate_env.sh (hunk 1 above)
Comment (OrangeBao) [reply hack/k8s-in-k8s/generate_env.sh:72]: This script is used to assist in generating environment variables, and its results need to be adjusted according to specific circumstances.

**Pull request merged**